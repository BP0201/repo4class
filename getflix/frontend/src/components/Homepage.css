.Homepage a.link:hover {
  color: rgb(31, 176, 99);
}