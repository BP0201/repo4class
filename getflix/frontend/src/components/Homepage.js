import React, { useContext, useState, useEffect } from 'react';
import { Link } from 'react-router-dom'
import MovieAPI from '../apis/MovieApi';
import UserContext from '../UserContext';
import DataCard from './DataCard';

import './Homepage.css'

const Homepage = ({ add, remove }) => {
    const { currentUser } = useContext(UserContext);
    const [trending, setTrending] = useState([]);

    useEffect(function getTrendingOnMount() {
        async function getTrending() {
            if (currentUser) {
                setTrending(await MovieAPI.getTrending())
            }
        }
        getTrending()
    }, [currentUser])

    return (
        <div className='Homepage'>
            {currentUser ?
            <div className="container col-sm-6 col-md-12 col-lg-10">
                <h4 data-testid="trending-header" className='display-4 my-3 text-center'>
                    What's Hot <span aria-label="flame_emoji" role="img">ðŸ”¥ðŸ”¥</span>
                </h4>
                {trending.map(d => (
                    <DataCard
                        name={d.title || d.name}
                        key={d.id}
                        id={d.id}
                        poster={ d.poster_path }
                        release_date={d.release_date || d.first_air_date}
                        // genres={d.genres}
                        description={d.overview}
                        category={d.media_type}
                        add={add}
                        remove={remove}
                    />
                ))}
            </div>

            :

            <div className="container col-md-6 col-lg-12">

                <div className='row justify-content-center'>
                <div className="card col-4 my-5 shadow">
                    <div className="card-body">
                        <h3 className="card-title text-center">Already have an account?</h3>
                        <Link className="link" data-testid="login-link" to="/login">Click here to login</Link>
                        <br />
                        <span>Or <Link className='link' to="/signup">signup</Link> here.</span>
                    </div>
                </div>
                </div>

                <div className='row text-center'>
                    <div className='mt-5 col'>
                    <h1 className='Title'>
                        Getflix
                    </h1>
                    </div>
                </div>

                <div className='Intro text-center'>
                    <div className='row mt-5 mb-3'>
                        <h2 className='col'>Browse movies and shows from all over the world.</h2>
                    </div>
                    <div className='row'>
                        <p className='col'>The information available on this website comes from The Movie Database API.</p>
                    </div>
                </div>

            </div>
            }
        </div>
    )
}

export default Homepage;