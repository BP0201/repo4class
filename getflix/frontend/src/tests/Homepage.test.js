import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Homepage from '../components/Homepage';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './testUtils';

describe("Homepage", () => {
  it('matches snapshot when logged in', () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <UserProvider>
          <Homepage />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    // only shown when logged in
    expect(getByTestId("trending-header")).toBeInTheDocument();
  });

  it('matches snapshot when logged out', () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <UserProvider currentUser={null}>
          <Homepage />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    // <Link to="/login"/> only shown when logged out
    expect(getByTestId("login-link")).toBeInTheDocument();
  });
});