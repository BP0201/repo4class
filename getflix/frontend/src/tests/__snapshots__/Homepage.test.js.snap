// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container col-sm-6 col-md-12 col-lg-10"
    >
      <h4
        class="display-4 my-3 text-center"
        data-testid="trending-header"
      >
        What's Hot 
        <span
          aria-label="flame_emoji"
          role="img"
        >
          ðŸ”¥ðŸ”¥
        </span>
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Homepage matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="container col-md-6 col-lg-12"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="card col-4 my-5 shadow"
        >
          <div
            class="card-body"
          >
            <h3
              class="card-title text-center"
            >
              Already have an account?
            </h3>
            <a
              class="link"
              data-testid="login-link"
              href="/login"
            >
              Click here to login
            </a>
            <br />
            <span>
              Or 
              <a
                class="link"
                href="/signup"
              >
                signup
              </a>
               here.
            </span>
          </div>
        </div>
      </div>
      <div
        class="row text-center"
      >
        <div
          class="mt-5 col"
        >
          <h1
            class="Title"
          >
            Getflix
          </h1>
        </div>
      </div>
      <div
        class="Intro text-center"
      >
        <div
          class="row mt-5 mb-3"
        >
          <h2
            class="col"
          >
            Browse movies and shows from all over the world.
          </h2>
        </div>
        <div
          class="row"
        >
          <p
            class="col"
          >
            The information available on this website comes from The Movie Database API.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
